src/components/RegisterLogin/Login/Login.js

```diff
@@ -2,34 +2,31 @@ import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
 import Form from 'react-bootstrap/Form';
 import { withRouter } from 'react-router-dom';
-import { loginUser } from '../../../redux/actions'
 import { connect } from 'react-redux';
-import classnames from 'classnames';
-
+import { loginUser } from '../../../redux/actions'
 import './Login.css';
 
 class Login extends PureComponent {
-
   state = {
     email: '',
     password: '',
     errors: {}
-  }
+  };
 
   componentDidMount() {
-    if (this.props.isAuthenticated) {
-      this.props.history.push('/')
-    }
-  }
+    const {
+      isAuthenticated,
+      history,
+    } = this.props;
 
-  static getDerivedStateFromProps(nextProps, prevState) {
-    if (nextProps.isAuthenticated) {
-      this.props.history.push('/');
+    if (isAuthenticated) {
+      history.push('/')
     }
+  }
 
-    if (nextProps.errors) {
-      return { errors: nextProps.errors };
-    }
+  static getDerivedStateFromProps({ isAuthenticated, errors }, prevState) {
```

Comment:
I'm not actually sure `prevState` is used here. If it's not used, it shouldn't be listed as an arg.

Reply:
I think I got this from the Traversy tutorial and I have read about it elsewhere being best practice?  But yeah, I'm not too sure how nextProps works.